### PR TITLE
Request for comment: Initial sync status without batch size change

### PR DESCRIPTION
**Pull request by alxndrsn**
Re-opening #2032 without `batch_size` changes.

> I've hacked up a quick initial sync status feature in the about page. No translations or anything fancy like that, but it should allow tech leads to just wait for that to change from pending to complete before handing them to CHWs.
>
> Dev feedback in how it's done (angular who?) encouraged.
>
> Not ready for merge (need to add translations, fix tests, maybe rename some things).
>
> – @SCdF in https://github.com/medic/medic-webapp/pull/2032

### CODE REVIEW

File: static/js/services/db-sync.js [hunk 1]
```diff
@@ -97,24 +95,40 @@ var _ = require('underscore'),
           });
       };
 
-      return function() {
+      return function(replicateDoneListener) {
```
Comment (garethbowen) [inline static/js/services/db-sync.js]: The naming convention is "callback". Listener implies EventListener (to me) that is a different pattern entirely.

File: static/js/services/db-sync.js [hunk 2]
```diff
@@ -97,24 +95,40 @@ var _ = require('underscore'),
...
           .then(function(params) {
             replicate('from', {
               filter: 'erlang_filters/doc_by_place',
+              live: false,
               query_params: params
+            })
+            .on('complete', function() {
+              console.log('Initial sync complete in ' +
+                ((Date.now() - beforeInitialReplication) / 1000) +
+                ' seconds, starting replication listener');
```
Comment (garethbowen) [inline static/js/services/db-sync.js]: Always use $log where you can - it gives us more flexibility over logging.

File: templates/partials/about.html [hunk 3]
```diff
@@ -10,6 +10,17 @@
...
+      <h3>{{'initial.replication.title' | translate}}</h3>
+      <ul>
+        <li>{{'initial.replication.status' | translate}}: {{initialReplicationStatus}}</li>
+        <li ng-show="initialReplicationDuration">{{'initial.replication.duration' | translate}}: {{initialReplicationDuration / 1000 | number:1}} s</li>
+        <span ng-show="initialReplicationDataUsage">
+          <li>{{'android_app.data_usage.rx' | translate}}: {{initialReplicationDataUsage.rx | number}} B</li>
+          <li>{{'android_app.data_usage.tx' | translate}}: {{initialReplicationDataUsage.tx | number}} B</li>
+        </span>
+      </ul>
+    </div>
```
Comment (garethbowen) [inline templates/partials/about.html]: Use the translate directive, eg:

```
<li><span translate>android_app.data_usage.tx</span>: {{initialReplicationDataUsage.tx | number}} B</li>
```

Comment (garethbowen) [inline templates/partials/about.html]: Some of these translation keys aren't in medic-api, eg: "initial.replication.status". Add the default translations to messages-en.properties as part of this change.

**Comment by garethbowen** [comment]
Some comments, and some tests need fixing. Otherwise looks good.

**Comment by alxndrsn** [comment]
:+1: thanks for the review.  Didn't want to add translations before this was accepted, but will do at merge time.

**Comment by alxndrsn** [comment]
Updated at #2045 - rebased against latest `develop`.

**Pull request closed without merging**